### ISSUE TITLE
Details on accessibility testing

### ISSUE THREAD

**Issue by mayank99** [ariakit/ariakit#4027]
### Motivation

Since Ariakit is an accessibility-focused library, it would be great to have documentation around how this accessibility is validated. I've found some vague claims here and there (for example, it's great that every component links to the corresponding ARIA spec), but I feel like it's missing a more nuanced, systematic way of documenting accessibility details.

From ["Your Accessibility Claims Are Wrong, Unless…"](https://adrianroselli.com/2022/11/your-accessibility-claims-are-wrong-unless.html):

>  Don’t assert the thing works in screen readers (for example) and leave it at that. Tell me which screen readers (plural). Tell me which versions of those screen readers on which operating systems with which browsers and which versions of those browsers.
> 
> Tell me how you tested it, which commands, which interaction mode (sticking with screen readers here), what was announced, where they differed.

This kind of information will be extremely useful for users/developers to evaluate which Ariakit components/examples are well-tested and work well with assistive technology, which ones still have issues (found during testing), which ones are pending testing, etc. When a certain feature hasn't been recently tested in Ariakit, it would be a clear indicator for users that they need to spend extra time testing it on their end.

### Usage example

As an example, take a look at the ["Accessibility" sections](https://carbondesignsystem.com/components/overview/accessibility-status) of [Carbon design system](https://carbondesignsystem.com/). Every component page has this information available, and there is also a central page for a bird's eye view of all components.

Related, see the ["Research" sections](https://design-system.service.gov.uk/components/cookie-banner/#research-on-this-component) on component pages of [Gov.UK design system](https://design-system.service.gov.uk/).

See also [Radix UI](https://www.radix-ui.com/primitives) which documents things like [keyboard interactions](https://www.radix-ui.com/primitives/docs/components/select#accessibility) for complex components. 

### Requirements

- Documentation pages for components and examples should indicate how, if, and what parts of the code was tested with which assistive technologies (including version numbers and browser pairings). Any problems found during testing should be clearly listed.
  - For automated accessibility tests, it would be helpful to also link to the test source code from the documentation.
  - For manual tests, it would be helpful to have captioned videos showing how it works with different assistive technology.
- Optionally, each of these pages could have badges (similar to Carbon design system) indicating the testing status of that component/example. This would make it easier to glance; it would also be useful for contributors who want to add tests.

**Comment by diegohaz** [comment]
Some examples do include detailed notes on accessibility tests (e.g., [Combobox Tabs: Testing with screen readers](https://ariakit.org/examples/combobox-tabs#testing-with-screen-readers) and [Submenu with Combobox: Testing with screen readers](https://ariakit.org/examples/menu-nested-combobox#testing-with-screen-readers)). However, I agree they could benefit from more details, like browser and screen reader combinations. Producing videos is currently out of our budget, but we may eventually get there.

I don't think this information should determine whether or how much developers test their apps. When a component or example is used in context, the context can impact accessibility, even if it works in isolation. These notes should encourage developers to test more and report any issues they find.
